app/models/app_resource_monitor/hud_references_inspector.rb

```diff
@@ -0,0 +1,165 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# report on invalid HUD references
+class AppResourceMonitor::HudReferencesInspector
+  def self.enrollment_references
+    new.enrollment_references
+  end
+
+  def self.project_references
+    new.project_references
+  end
+
+  def self.client_references
+    new.client_references
+  end
+
+  def self.duplicate_ids
+    new.duplicate_ids
+  end
+
+  def duplicate_ids
+    [
+      Hmis::Hud::Affiliation,
+      Hmis::Hud::Assessment,
+      Hmis::Hud::CeParticipation,
+      Hmis::Hud::Client,
+      Hmis::Hud::CurrentLivingSituation,
+      Hmis::Hud::CustomAssessment,
+      Hmis::Hud::CustomCaseNote,
+      Hmis::Hud::CustomClientAddress,
+      Hmis::Hud::CustomClientContactPoint,
+      Hmis::Hud::CustomClientName,
+      Hmis::Hud::CustomService,
+      Hmis::Hud::Disability,
+      Hmis::Hud::EmploymentEducation,
+      Hmis::Hud::Enrollment,
+      Hmis::Hud::Event,
+      Hmis::Hud::Exit,
+      Hmis::Hud::Funder,
+      Hmis::Hud::HealthAndDv,
+      Hmis::Hud::HmisParticipation,
+      Hmis::Hud::IncomeBenefit,
+      Hmis::Hud::Inventory,
+      Hmis::Hud::ProjectCoc,
+      Hmis::Hud::Service,
+      Hmis::Hud::YouthEducationStatus,
+    ].flat_map do |model|
```

Comment:
I think we should lean on some existing lists if possible, and not on the HMIS driver (if possible).
We might want to make a method somewhere in the HMIS driver that could return the custom data, or maybe we should just add it to the `by_hud_filename` method.  I'm not 100% sure where that's used or if it would matter, but I'd like to keep the lists of available classes from growing since they change every couple years and tracking them down can be a pain.
```suggestion
    hud_models = GrdaWarehouse::Hud.models.by_hud_filename.values
    if RailsDrivers.loaded.include?(:hmis)
      hud_models += [
        Hmis::Hud::CustomAssessment,
        Hmis::Hud::CustomCaseNote,
        Hmis::Hud::CustomClientAddress,
        Hmis::Hud::CustomClientContactPoint,
        Hmis::Hud::CustomClientName,
        Hmis::Hud::CustomService,
       ]
     end
     hud_models.sort.flat_map do |model|
```